src/ngx_http_lua_shdict.h

```diff
@@ -11,6 +11,23 @@
 #include "ngx_http_lua_common.h"
 
 
+struct ngx_http_lua_shdict_ctx_s;
+typedef struct ngx_http_lua_shdict_ctx_s ngx_http_lua_shdict_ctx_t;
+typedef void (* memory_rlock)(ngx_http_lua_shdict_ctx_t *);
+typedef void (* memory_wlock)(ngx_http_lua_shdict_ctx_t *);
+typedef void (* memory_unlock)(ngx_http_lua_shdict_ctx_t *);
+typedef void (* memory_lock_downgrade)(ngx_http_lua_shdict_ctx_t *);
```

Comment:
Also, it's bad to pollute the global namespace with unqualified names like `memory_rlock`. Note that all the nginx global symbols (and also file-scoped static ones) use prefixes like `ngx_http_xxx` to avoid name space pollution.